### PR TITLE
Sendgrid API v3

### PR DESCRIPTION
**Pull request by dbulic**
Pure JSON, custom variables, tests.

My first ever contribution to open-source! :)

### CODE REVIEW

**Comment by stevedomin** [comment]
Thanks so much for that @dbulic!!

Will try to have a proper look tonight, other tomorrow.

**Comment by barisbalic** [comment]
Looks good, if you squash the commits we'll get this merged.

**Comment by dbulic** [comment]
I did, but now it shows 7 commits to merge - I hope I did it properly. If not, please tell me what to do exactly.

File: lib/swoosh/adapters/sendgrid.ex [hunk 1]
```diff
@@ -2,7 +2,7 @@ defmodule Swoosh.Adapters.Sendgrid do
   @moduledoc ~S"""
   An adapter that sends email using the Sendgrid API.
 
-  For reference: [Sendgrid API docs](https://sendgrid.com/docs/API_Reference/Web_API/mail.htm://sendgrid.com/docs/API_Reference/Web_API/mail.html)
+  For reference: [Sendgrid API docs](https://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/index.html://sendgrid.com/docs/API_Reference/Web_API_v3/Mail/index.html)
```
Comment (stevedomin) [inline lib/swoosh/adapters/sendgrid.ex]: Looks like this link is broken (they were before), can you update it?

File: lib/swoosh/adapters/sendgrid.ex [hunk 2]
```diff
@@ -46,63 +45,58 @@ defmodule Swoosh.Adapters.Sendgrid do
...
-    |> prepare_to(email)
-    |> prepare_cc(email)
-    |> prepare_bcc(email)
+    |> prepare_personalizations(email)
     |> prepare_subject(email)
-    |> prepare_html_body(email)
-    |> prepare_text_body(email)
+    |> prepare_content(email)
     |> prepare_reply_to(email)
+    |> Poison.encode!
```
Comment (stevedomin) [inline lib/swoosh/adapters/sendgrid.ex]: Can you move this to the `deliver/2` function. Like this:

`body = email |> prepare_body() |> Poison.encode!`

That way we are consistent with the other adapters, and this function is only concerned about building a representation of the body, not how it's encoded.

**Comment by stevedomin** [comment]
@dbulic I added two minor comments

Once these are fixed I'll squash and merge. Thanks!

**Comment by dbulic** [comment]
@stevedomin, thanks! I did as you asked, hope things are good to go now.

**Comment by stevedomin** [comment]
Thanks so much @dbulic!

**Pull request merged**